src/Entity/BaseUser.php

```diff
@@ -46,9 +46,9 @@ public function preUpdate(): void
     public static function getGenderList()
     {
         return [
-            'gender_unknown' => UserInterface::GENDER_UNKNOWN,
-            'gender_female' => UserInterface::GENDER_FEMALE,
-            'gender_male' => UserInterface::GENDER_MALE,
+            UserInterface::GENDER_UNKNOWN => 'gender_unknown',
```

Comment:
This is a BC break. You should flip this values in your form type

Reply:
The implication is that this was a BC break too, right: https://github.com/sonata-project/SonataUserBundle/commit/c69d51bdd72c50340aa04c9b14a95b18e8600336#diff-0b70ecd19ebc26172322781dd1061c2fL47 ?

Reply:
Oh but it was in a major PR, so, completely normal.

Reply:
I was thinking about BC,, however w/o this change, it seems that it does not work (at least on 4.0.2), so I was created this simple fix.
I've extended this form type, and flipped it myself, however I think it should be in upcoming version (otherwise you will get some ugly sql error  - gender type is varchar(1) -> and it try to save `gender_unknown`

Reply:
> it seems that it does not work (at least on 4.0.2), so I was created this simple fix.

The problem is not the method in the entity itself. Your problem is probably in your FormType

Reply:
@core23
It caused because of https://github.com/sonata-project/SonataCoreBundle/blob/master/src/Form/Type/BaseStatusType.php
with combination with https://github.com/sonata-project/SonataUserBundle/blob/4.x/src/Resources/config/form.xml#L7
and 
https://github.com/sonata-project/SonataUserBundle/blob/4.x/src/Form/Type/UserGenderListType.php#L18

`UserGenderListType` extends `BaseStatusType` which unfortunatelly has 4th argument set to `null` which cause that values returned from `getGenderList` method are flipped

Reply:
what about tell the BaseStatusType to not flip those choices?

Reply:
it was my 1st idea, however It says that 4th argument is depreciated, so it has to be changed again in v4

Reply:
@core23 what we should do here? If we cannot override the fourth parameter (deprecated) we need to flip the choices right?

Reply:
Create a new `AbstractStatusType`? https://github.com/sonata-project/SonataCoreBundle/pull/499#issuecomment-357534483

Reply:
For this issue, we must keep this mapping and can't flip this right now. Please change the branch for this PR to master an leave an upgrade note.

Reply:
Should I add note to `UPGRADE-4.x.md` or is PR description itself?

Reply:
IMO it is a better solution to create a new AbstactStatus time. And even better, why do we need a type that is only extending the choiceType. Can you try instead changing the admin to use ChoiceType directly? The choices are on  a static method, so you can access them.